crates/ide/src/ide/assists/add_to_lambda.rs

```diff
@@ -0,0 +1,84 @@
+//! add an undefined name to the top-level lambda
+use super::{AssistKind, AssistsCtx};
+use crate::def::Name;
+use crate::TextEdit;
+use syntax::ast::{self, AstNode};
+use syntax::{TextRange, TextSize};
+
+pub(super) fn add_to_lambda(ctx: &mut AssistsCtx<'_>) -> Option<()> {
+    let name = ctx.covering_node::<ast::Ref>()?.token()?;
+    let name = name.text();
+
+    // the whole file should be a lambda
+    let pat = ast::Lambda::cast(ctx.ast.syntax().first_child()?)?
+        .param()?
+        .pat()?;
+
+    // name should be undefined
+    if ctx
+        .db
+        .module(ctx.frange.file_id)
+        .names()
+        .any(|(_, Name { text, .. })| text == name)
+    {
+        return None;
+    };
+
+    let (pos, insert) = if let Some(field) = pat.fields().last() {
+        let pos = field.syntax().text_range().end();
+        let left = pos - TextSize::from(1);
+        // insert before the space if the field ends with a space
+        (
+            if ctx.ast.syntax().text().char_at(left) == Some(' ') {
+                left
+            } else {
+                pos
+            },
+            format!(", {name}"),
+        )
+    } else if let Some(ellipsis) = pat.ellipsis_token() {
+        (ellipsis.text_range().start(), format!("{name}, "))
+    } else {
+        (
+            pat.syntax().text_range().end() - TextSize::from(1),
+            format!("{name} "),
+        )
+    };
+
+    ctx.add(
+        "add_to_lambda",
+        format!("Add {name} to the top-level lambda"),
```

Comment:
The mod name should be tweaked. `add_to_lambda` is too ambiguous.
```suggestion
        "add_to_top_level_lambda_param",
        format!("Add `{name}` to the top-level lambda parameter"),
```